File: go/hamming/hamming.go
package hamming

import "errors"

// Distance takes two strings of equal length returns the number of differences
// This number is the hamming distance
// if the lengths of a and b are different an error is returned
func Distance(a, b string) (int, error) {
	distance := 0
	if len(a) != len(b) {
		return distance, errors.New("left and right strands do not match")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}

	return distance, nil
}

File: java/hamming/src/main/java/Hamming.java
class Hamming {

	char[] left;
	char[] right;

	Hamming(String leftStrand, String rightStrand) {
		if (leftStrand == "" && rightStrand != "")
			throw new IllegalArgumentException("left strand must not be empty.");
		if (rightStrand == "" && leftStrand != "")
			throw new IllegalArgumentException("right strand must not be empty.");
		if (leftStrand.length() != rightStrand.length())
			throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");

		left = leftStrand.toCharArray();
		right = rightStrand.toCharArray();
	}

	int getHammingDistance() {
		int distance = 0;
		for (int i = 0; i < left.length; i++) {
			if (left[i] != right[i])
				distance++;
		}
		return distance;
	}

}
